### PR TITLE
Seeded RNG

### PR DESCRIPTION
**Pull request by Cosine256**
This PR updates all RNG calls to use the `prng` library. Seeded runs will now always generate the same sequence of levels. I'm not 100% certain that I used the "correct" `PRNG_CLASS` for everything, but I think my choices are mostly accurate, and this doesn't really matter to players.

### CODE REVIEW

File: lib/options.lua [hunk 1]
```diff
@@ -106,7 +106,7 @@ end)
 local function draw_warp_button(ctx, name, world, level, theme, world_state)
     if ctx:win_button(name) then
         if warp_reset_run then
-            state.quest_flags = QUEST_FLAG.RESET
+            state.quest_flags = set_flag(state.quest_flags, QUEST_FLAG.RESET)
```
Comment (SuperNinjaFat) [inline lib/options.lua:109]: Might want to include this fix in the description

**Review by SuperNinjaFat — APPROVED** [review]
Nice work!

File: lib/options.lua (hunk 1 above)
Comment (SuperNinjaFat) [reply lib/options.lua:109]: Ah ok I see that this was a part of the commit, "Fix seed by clearing warps"

File: lib/common.lua [hunk 2]
```diff
@@ -65,8 +65,8 @@ end
 --Use _, value = next(t)
 --function module.TableFirstValue(t)
 
-function module.TableCopyRandomElement(tbl)
-	return module.TableCopy(tbl[math.random(#tbl)])
+function module.TableCopyRandomElement(tbl, prng_class)
```
[the remark this answers is not in the document]
Comment (Cosine256) [reply lib/common.lua:68]: I think it's safe. We use that function only a few times and I added the new parameter to all of those usages.

**Pull request merged**